cmd/awl/windows_embeds.go

```diff
@@ -20,9 +25,38 @@ func init() {
 		return
 	}
 	wintunPath := filepath.Join(filepath.Dir(ex), "wintun.dll")
-	err = os.WriteFile(wintunPath, wintunDLL, 664)
+	writeWintun := func() {
+		err = os.WriteFile(wintunPath, wintunDLL, 664)
+		if err != nil {
+			fmt.Printf("error: write wintun.dll file: %v\n", err)
+		}
+	}
+
+	// wintun does not exist
+
+	if _, err := os.Stat(wintunPath); errors.Is(err, os.ErrNotExist) {
+		writeWintun()
+		return
+	}
+
+	// wintun exist
+
+	existedWintun, err := os.Open(wintunPath)
 	if err != nil {
-		fmt.Printf("error: write wintun.dll file: %v\n", err)
+		fmt.Printf("error: read wintun.dll file: %v\n", err)
 		return
 	}
+	defer func() {
+		err := existedWintun.Close()
+		if err != nil {
+			fmt.Printf("error: close read wintun.dll file: %v\n", err)
+		}
+	}()
+	equal, err := libs.StreamsEqual(bytes.NewReader(wintunDLL), bufio.NewReader(existedWintun))
+	if err != nil {
+		fmt.Printf("error: compare wintun.dll files: %v\n", err)
+	}
+	if !equal {
+		writeWintun()
+	}
```

Comment:
I think this is a good time to move all this code (except embed) to a shared package across awl and awl-tray. How about `cli` package? And probably move `StreamsEqual` func too

Reply:
I don't fully understand which part of code you want to move)
So, my options:
1) Only write Wintun
2) Change init function to callable function and call it, obviously) 

I think that lib package with helpful little functions, grouped by functionality, sooner or later it had to appear, this is the fate of any big project) IMHO, so I would prefer not to delete the lib directory

Reply:
3-rd option. Move content of init function to some callable function, but still use init and call it in init

Reply:
I was thinking about option 3.

Utils package is a huge anti-pattern, I prefer to avoid it when possible. In this case we use this function only in one place, which means we don't need a new package yet. In the future we could have something like `ioutil` if needed

Reply:
kk, but i use not ` cli`, but new `embeds` package

Reply:
You'd also need to change path to `wintun.dll` in `build.sh` script